src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1412,6 +1412,11 @@ MasterExtendedOpNode(MultiExtendedOp *originalOpNode,
 			column->varoattno = walkerContext->columnId;
 			walkerContext->columnId++;
 
+			if (column->vartype == RECORDOID)
```

Comment:
Should check if we're sometimes having redundant entry into this logic (ie having called `BlessRecordExpression` already earlier in planning), perhaps add a check to skip blessing when vartypmod already set

Reply:
At least within the test suite the two blessing paths are mutually exclusive